src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
```

Comment:
You can put the condition about "withChildren" inside the filter() function, like follow:

```suggestion
  const women = people.filter((person) => {
    const isWoman = person.sex === "f";

    if (!withChildren) return isWoman;

    const hasChildren = people.some(
      (children) => children.mother === person.name
    );

    return isWoman && hasChildren;
  });
```

Changing the variables names to make it more clear